### ISSUE TITLE
Interpolation of pos_embed for CroCov2

### ISSUE THREAD

**Issue by HeimbergM** [naver/croco#38]
Hello! 
I am looking for a way to interpolate the pos_embed of the CroCov2 model as i need a higher resolution of the encoded feature maps!

It is not as trivial for the RoPE embedding  and i am not sure if my approach is right:

- I work with the patch embedding and changed the stride to 4x4 while keeping the patch size of 16x16.
- Therefore the number of patches is 53x53 in contrast to 14x14 as before!
- I changed one line in RoPE, where  I divide t by 4 so it ranges from 0 to 14 but keeps the length of 53:

`        def get_cos_sin(self, D, seq_len, device, dtype):
            if (D,seq_len,device,dtype) not in self.cache:
                inv_freq = 1.0 / (self.base ** (torch.arange(0, D, 2).float().to(device) / D))
                
            -->t = torch.arange(seq_len, device=device, dtype=inv_freq.dtype) / 4
                
                freqs = torch.einsum("i,j->ij", t, inv_freq).to(dtype)
                freqs = torch.cat((freqs, freqs), dim=-1)
                cos = freqs.cos() # (Seq, Dim)
                sin = freqs.sin()
                self.cache[D,seq_len,device,dtype] = (cos,sin)
                return self.cache[D,seq_len,device,dtype]`

By any chance do you know if this has a chance to work out on a pretrained model?
